pkg/services/apiserver/aggregator/aggregator.go

```diff
@@ -128,15 +128,20 @@ func CreateAggregatorConfig(commandOptions *options.Options, sharedConfig generi
 		},
 		ExtraConfig: aggregatorapiserver.ExtraConfig{
 			DisableRemoteAvailableConditionController: true,
-			ProxyClientCertFile:                       commandOptions.KubeAggregatorOptions.ProxyClientCertFile,
-			ProxyClientKeyFile:                        commandOptions.KubeAggregatorOptions.ProxyClientKeyFile,
 			// NOTE: while ProxyTransport can be skipped in the configuration, it allows honoring
 			// DISABLE_HTTP2, HTTPS_PROXY and NO_PROXY env vars as needed
 			ProxyTransport:  createProxyTransport(),
 			ServiceResolver: serviceResolver,
 		},
 	}
 
+	if commandOptions.KubeAggregatorOptions.LegacyClientCertAuth {
+		// NOTE: the availability controller below is a bit different and uses the cert/key pair regardless
+		// of the legacy bool, this is because we are still using that for discovery requests
+		aggregatorConfig.ExtraConfig.ProxyClientCertFile = commandOptions.KubeAggregatorOptions.ProxyClientCertFile
+		aggregatorConfig.ExtraConfig.ProxyClientKeyFile = commandOptions.KubeAggregatorOptions.ProxyClientKeyFile
```

Comment:
i might be missing something with the new option, but it seems like we could not set the existing options and achieve the same result?

Reply:
The main issue is that we need to set these options for discovery client, but not for the http.Handler which is reverse proxying upstream, does that make sense?

That means a second bool must be introduced to determine whether its old behavior (use certs everywhere) vs new behavior (use certs only for discovery).

Of course, based on discussion, we could determine we don't want such a hybrid approach, but @ferruvich was hoping we could make the change to discovery in a subsequent PR and first test whether the reverse proxying process can use the bespoke token auth we are introducing in enterprise.